src/Angor/Shared/Services/IndexerService.cs

```diff
@@ -225,5 +239,40 @@ public async Task<string> GetTransactionHexByIdAsync(string transactionId)
 
             return info;
         }
+
+        public async Task<string> GetGenesisBlockHashAsync(string indexerUrl)
```

Comment:
I suggest that this method will also check if the indexer is online and also return the genesis hash.
then you can avoid the hack of adding to settings and then removing.
The code to check inline is simple it is just 
` var response = await _httpClient.GetAsync($"{uri}api/stats/heartbeat");`